# HG changeset patch
# User Kurt Smith <kwsmith1@wisc.edu>
# Date 1237319878 18000
# Node ID f225a23de8ff8390a3b78e47f10a9ba628a1a3ad
# Parent  e2365a6d00b8d0cdb665699e840a8f052d471e37
Fix 135 -- scope objects have return_type set to None by default

diff -r e2365a6d00b8 -r f225a23de8ff Cython/Compiler/Symtab.py
--- a/Cython/Compiler/Symtab.py	Tue Mar 17 07:32:51 2009 +0100
+++ b/Cython/Compiler/Symtab.py	Tue Mar 17 14:57:58 2009 -0500
@@ -235,6 +235,7 @@ class Scope(object):
         self.pystring_entries = []
         self.buffer_entries = []
         self.control_flow = ControlFlow.LinearControlFlow()
+        self.return_type = None
         
     def start_branching(self, pos):
         self.control_flow = self.control_flow.start_branch(pos)
diff -r e2365a6d00b8 -r f225a23de8ff tests/errors/return_outside_function_T135.pyx
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/tests/errors/return_outside_function_T135.pyx	Tue Mar 17 14:57:58 2009 -0500
@@ -0,0 +1,8 @@
+return 'bar'
+class A:
+    return None
+
+_ERRORS = u'''
+1:0: Return not inside a function body
+3:4: Return not inside a function body
+'''
# HG changeset patch
# User Kurt Smith <kwsmith1@wisc.edu>
# Date 1237320614 18000
# Node ID f153d2da27b08c0fb35827d6b545007fe15cd6f3
# Parent  f225a23de8ff8390a3b78e47f10a9ba628a1a3ad
Fix 135 -- small addition to test case.

diff -r f225a23de8ff -r f153d2da27b0 tests/errors/return_outside_function_T135.pyx
--- a/tests/errors/return_outside_function_T135.pyx	Tue Mar 17 14:57:58 2009 -0500
+++ b/tests/errors/return_outside_function_T135.pyx	Tue Mar 17 15:10:14 2009 -0500
@@ -1,8 +1,12 @@ return 'bar'
 return 'bar'
 class A:
+    return None
+
+cdef class B:
     return None
 
 _ERRORS = u'''
 1:0: Return not inside a function body
 3:4: Return not inside a function body
+6:4: Return not inside a function body
 '''
